File: java/src/main/java/QuestionSix.java
public class QuestionSix {

	public static int[][] transpose(int[][] matrix) {
		int row = matrix.length;
		int column = matrix[0].length;

		int[][] transposedMatrix = new int[column][row];

		for (int i = 0; i < row; i++) {
			for (int j = 0; j < column; j++) {
				transposedMatrix[j][i] = matrix[i][j];
			}
		}
		return transposedMatrix;
	}
}

File: go/questions/question_six.go
package questions

func Transpose(matrix [][]int) [][]int {
	row := len(matrix)
	column := len(matrix[0])
	transposedMatrix := make([][]int, column)

	for i := range transposedMatrix {
		transposedMatrix[i] = make([]int, row)
	}

	for i := 0; i < row; i++ {
		for j := 0; j < column; j++ {
			transposedMatrix[j][i] = matrix[i][j]
		}
	}
	return transposedMatrix
}
